org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/GradleBuild.java

```diff
@@ -41,42 +32,15 @@
     /**
      * Attempts to synchronize the build with the workspace.
      * <p/>
-     * The synchronization happens asynchronously. In case of a failure, the user will be notified
-     * once after the build has finished.
-     * <p/>
-     * This is equivalent to calling {@code synchronize(NewProjectHandler.NO_OP)}
-     */
-    void synchronize();
-
-    /**
-     * Attempts to synchronize the build with the workspace.
-     * <p/>
-     * The synchronization happens asynchronously. In case of a failure, the user will be notified
-     * once the build has finished.
-     * <p/>
-     * This is equivalent to calling {@code synchronize(newProjectHandler, AsyncHandler.NO_OP)}
+     * The synchronization happens synchronously. In case of a failure, the method throws a
+     * {@link CoreException} which contains the necessary status and error message about the
+     * failure.
      *
      * @param newProjectHandler how to handle newly added projects
+     * @param tokenSource the cancellation token source
+     * @see org.eclipse.buildship.core.util.progress.ToolingApiStatus
      */
-    void synchronize(NewProjectHandler newProjectHandler);
-
-    /**
-     * Attempts to synchronize the build with the workspace.
-     * <p/>
-     * The synchronization happens asynchronously. In case of a failure, the user will be notified
-     * once the build has finished.
-     *
-     * @param newProjectHandler how to handle newly added projects
-     * @param initializer an initializer to run before synchronization, e.g. to create a new project
-     */
-    void synchronize(NewProjectHandler newProjectHandler, AsyncHandler initializer);
-
-    /**
-     * Returns {@code true} if a synchronization job is already running for the same root project.
-     *
-     * @return true if a synchronization is running.
-     */
-    boolean isSyncRunning();
+    void synchronize(NewProjectHandler newProjectHandler, CancellationTokenSource tokenSource, IProgressMonitor monitor) throws Exception;
```

Comment:
⭕️ Shouldn't this only throw `CoreException` to be consistent with other Eclipse APIs?

Reply:
It makes sense, I've changed it.